src/main/java/picard/analysis/CompareMetrics.java

```diff
@@ -267,22 +267,26 @@ protected SimpleResult compareMetricValues(final Object value1, final Object val
             if (value1 instanceof Number) {
                 double numValue1 = ((Number) value1).doubleValue();
                 double numValue2 = ((Number) value2).doubleValue();
-                double relativeChange = (numValue1 == 0 && numValue2 == 0) ? 0 : (numValue1 == 0) ? Double.MAX_VALUE : (numValue1 - numValue2) / numValue2;
-                if (relativeChange != 0) {
+                double absoluteChange = 0;
+                if (!Double.isNaN(numValue1) || !Double.isNaN(numValue2)) {
+                    absoluteChange = numValue2 - numValue1;
+                }
+                if (absoluteChange != 0) {
+                    double relativeChange = (numValue1 == 0 && numValue2 == 0) ? 0 : (numValue1 == 0) ? Double.MAX_VALUE : absoluteChange / numValue1;
```

Comment:
Under what conditions is it possible for `absoluteChange` to be non-zero and for `(numValue1 == 0 && numValue2 == 0)` to be true? Are this trying to avoid dividing by zero?

Also this code now divides by `numValue1` and before it divided by `numValue2`, is that a fix?

Reply:
Yes, that's to avoid the divide by zero.
I've changed the code to be comparison of the second metric, relative to the first metric, so that's why the denominator has changed.
change = (value2 - value1) / value1

Reply:
OK, so you only need to check `numValue1 == 0`. Dividing zero by anything (except NaN) is zero.

Reply:
That explains the second `numValue1 == 0` check. But I don't see how `(numValue1 == 0 && numValue2 == 0)` can be true if `absoluteChange` is not zero.

Reply:
Probably true, the problem with doing lots of incremental changes.

Reply:
And it's really not even a problem if the numerator is 0.